Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -168,22 +168,48 @@ namespace Babylon::Polyfills::Internal
         }
     }
 
-    void XMLHttpRequest::Abort(const Napi::CallbackInfo&)
+    void XMLHttpRequest::Abort(const Napi::CallbackInfo& info)
     {
-        m_request.Abort();
+        try
+        {
+            m_request.Abort();
+        }
+        catch (const std::exception& ex)
+        {
+            throw Napi::Error::New(info.Env(), ex.what());
+        }
     }
 
     void XMLHttpRequest::Open(const Napi::CallbackInfo& info)
     {
-        m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
-        SetReadyState(ReadyState::Opened);
+        try
+        {
+            m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
+            SetReadyState(ReadyState::Opened);
+        }
+        catch (const std::exception& ex)
+        {
+            throw Napi::Error::New(info.Env(), ex.what());
+        }
     }
 
-    void XMLHttpRequest::Send(const Napi::CallbackInfo& /*info*/)
+    void XMLHttpRequest::Send(const Napi::CallbackInfo& info)
     {
-        m_request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [this]() {
-            SetReadyState(ReadyState::Done);
-        });
+        try
+        {
+            m_request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [env{info.Env()}, this](arcana::expected<void, std::exception_ptr> result) {
+                if (result.has_error())
+                {
+                    throw Napi::Error::New(env, result.error());
+                }
+
+                SetReadyState(ReadyState::Done);
+            });
+        }
+        catch (const std::exception& ex)
+        {
+            throw Napi::Error::New(info.Env(), ex.what());
```

Comment:
I'm not quite following the distinction between when you pass in the actual `exception_ptr` vs the `what()` string. Can you explain?

Reply:
I'm not sure I understand the question. There are two separate things that can throw exceptions in this code.
1. The synchronous part of SendAsync may throw an exception.
2. The continuation may be in an exceptional state.

Does that answer the question? 🤔